### PR TITLE
Fix: Bar chart -> Support mix of positive + negative values

### PR DESCRIPTION
**Pull request by christophby**
A mix of _positive_ and _negative_ values in the stacked bar chart put the whole bar below the x-axis.
This PR allows a stacked bar chart to have _positive_ and _negative_ values in the same stack and renders it correctly.

With this stack data
```typescript
const stacks: stackItemType[] = [
    {
      // Mix of positive and negative values
      label: 'Nov "23',
      stacks: [
        {
          value: 20,
          color: '#35aa0a',
        },
        {
          value: -5,
          color: '#ffa600',
        },
        {
          value: -6,
          color: '#dc25bb',
        },
        {
          value: 10,
          color: '#003f5c',
        },
      ],
    },
    // Only positive values
    {
      label: 'Dez "23',
      stacks: [
        {
          value: 8,
          color: '#35aa0a',
        },
        {
          value: 1,
          color: '#ffa600',
        },
        {
          value: 7,
          color: '#dc25bb',
        },
      ],
    },
    // Only negative values
    {
      label: 'Dez "23',
      stacks: [
        {
          value: -5,
          color: '#35aa0a',
        },
        {
          value: 0,
          color: '#ffa600',
        },
        {
          value: -3,
          color: '#dc25bb',
        },
      ],
    },
  ];
```

### Before
image

### After (correct visualization)
* Move negative values below the x-axis
* Move bar labels below the bars
image

connected to https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/issues/261

### CODE REVIEW

File: src/BarChart/RenderStackBars.tsx [hunk 1]
```diff
@@ -136,22 +136,25 @@ const RenderStackBars = (props: Props) => {
...
       <View
         style={[
           {
             width:
               (item.stacks[0].barWidth || props.barWidth || 30) + spacing / 2,
             position: 'absolute',
-            bottom: rotateLabel ? -40 : -6 - xAxisTextNumberOfLines * 18,
+            bottom: rotateLabel
+              ? -40
+              : -6 - xAxisTextNumberOfLines * 18 + lowestBarPosition,
```
Comment (christophby) [inline src/BarChart/RenderStackBars.tsx:149]: Adding `lowestBarPosition` here to move bar label below the bar.
Otherwise the labels would be on top of the bars (in case of negative values)

File: src/Components/BarAndLineChartsWrapper/index.tsx [hunk 2]
```diff
@@ -324,8 +324,8 @@ const BarAndLineChartsWrapper = (props: BarAndLineChartsWrapperTypes) => {
           ? (props.width ? -98 - endSpacing : -75 - endSpacing) -
```
Comment (Abhinandan-Kushwaha) [inline src/Components/BarAndLineChartsWrapper/index.tsx:324]: Plz undo these indent changes.

**Comment by Abhinandan-Kushwaha** [comment]
Thanks @christophby for this much-needed CL. I will merge this before publishing the next version of the library (hopefully by Sunday)

File: src/Components/BarAndLineChartsWrapper/index.tsx (hunk 2 above)
Comment (christophby) [reply src/Components/BarAndLineChartsWrapper/index.tsx:324]: sure, sorry

Comment (christophby) [reply src/Components/BarAndLineChartsWrapper/index.tsx]: done

File: src/BarChart/RenderStackBars.tsx [hunk 3]
```diff
@@ -136,22 +136,25 @@ const RenderStackBars = (props: Props) => {
   }
   const disablePress = props.disablePress || false;
   const renderLabel = (label: String, labelTextStyle: any) => {
+    const lowestBarPosition = getLowestPosition();
```
Comment (Abhinandan-Kushwaha) [inline src/BarChart/RenderStackBars.tsx:139]: Plz define **_getLowestPosition()_** before using it. I mean move the function definition above this statement.

File: src/BarChart/RenderStackBars.tsx [hunk 4]
```diff
@@ -165,15 +168,45 @@ const RenderStackBars = (props: Props) => {
   };
 
   const getPosition = (index: number) => {
-    let position = 0;
+    /* Returns bottom position for stack item
+       negative values are below origin (-> negative position) */
+    const height = getBarHeight(
```
Comment (Abhinandan-Kushwaha) [inline src/BarChart/RenderStackBars.tsx:173]: Plz define **_getBarHeight_** before using it.

File: src/BarChart/RenderStackBars.tsx (hunk 3 above)
Comment (christophby) [reply src/BarChart/RenderStackBars.tsx:139]: I will. 
It would be super helpful if you could provide a runnable version of this project for contributers ,including prettier / eslint / dependencies etc.
Currently, if you checkout the repo, _npm install_ fails and you cannot run the examples out of the box (e. g. `yarn ios`) + the code formatters don't work properly.

Comment (Abhinandan-Kushwaha) [reply src/BarChart/RenderStackBars.tsx:139]: Thanks @christophby for your valuable suggestion.
AFAIK, the only change required to make the project runnable is to move `react` and `react-native` from _peerDependencies_ to _dependencies_ in `package.json`.

If you had to make some other changes to run the project, do let me know.

File: src/BarChart/RenderStackBars.tsx [hunk 5]
```diff
@@ -293,17 +326,15 @@ const RenderStackBars = (props: Props) => {
                 borderRadius ??
                 stackBorderBottomRightRadius ??
                 stackBorderRadius,
-              overflow: 'hidden',
```
Comment (Abhinandan-Kushwaha) [inline src/BarChart/RenderStackBars.tsx:296]: @christophby Was this change intentional? `stackBorderRadius` is not working after removing this.

**Review by Abhinandan-Kushwaha — APPROVED** [review]

File: src/BarChart/RenderStackBars.tsx (hunk 5 above)
Comment (Abhinandan-Kushwaha) [reply src/BarChart/RenderStackBars.tsx:296]: So this change is needed. I have merged it. To let `stackBorderRadius` work, I have added this property conditionally.

Comment (christophby) [reply src/BarChart/RenderStackBars.tsx:296]: Yes, it is. Thanks a lot
